jobserv/api/worker.py

```diff
@@ -39,11 +47,37 @@ def wrapper(*args, **kwargs):
         if not key:
             return jsendify("No Authorization header provided", 401)
         parts = key.split(" ")
-        if len(parts) != 2 or parts[0] != "Token":
+        if len(parts) != 2 or parts[0] not in ("Token", "Bearer"):
             return jsendify("Invalid Authorization header", 401)
-        worker = get_or_404(Worker.query.filter_by(name=kwargs["name"], deleted=False))
-        if not worker.validate_api_key(parts[1]):
-            return jsendify("Incorrect API key for host", 401)
+
+        if parts[0] == "Bearer":
+            try:
+                w = worker_from_jwt(parts[1])
+            except PyJWTError as e:
+                return jsendify(str(e), 401)
+
+            if w.name != kwargs["name"]:
+                # worker can only access its self
+                return jsendify("Not found", 404)
```

Comment:
Shouldn't it be 401?

Reply:
probably. i think i should do a change before this that locks down access to listing/viewing workers. then this work can be done more sanely.